data/remnant/remnant outfits.txt

```diff
@@ -276,6 +277,136 @@ outfit "EMP Torpedo Bay"
 		"missile strength" 80
 	description "Electromagnetic pulse weapons were developed by the Remnant during the early days of their colony, when they were living in fear that the Alphas would overrun human space and expand beyond it."
 
+
+
+outfit "Teciimach Canister"
+	plural "Teciimach Canisters"
+	category "Ammunition"
+	licenses
+		Remnant
+	cost 5500
+	thumbnail "outfit/emp torpedo"
+	"mass" 1
+	"teciimach canister capacity" -1
+	description "This canister contains everything required to assemble a Teciimach torpedo. These weapons emit strong electromagnetic pulses which interfere with a ship's generators and neutralize a portion of their energy."
+	description "Unlike the previous generation of 'EMP torpedoes', the Teciimach torpedoes synchronize themselves with their mothership in order to avoid impacting their allies."
+
+outfit "Teciimach Canister Rack"
+	plural "Teciimach Canister Racks"
+	category "Ammunition"
+	licenses
+		Remnant
+	cost 98000
+	thumbnail "outfit/emp rack"
+	"mass" 5
+	"outfit space" -9
+	"energy capacity" 500
+	"teciimach canister capacity" 9
+	ammo "Teciimach Canister"
+	description "The Teciimach Canister rack is used to store extra sets of components, both to expand the capacity of a ship, and on carriers for resupplying fighters."
+	description "	In addition to its ammunition capacity, the storage bay can also contain a small amount of charge, enough to let an otherwise battery-less ship assemble Teciimach torpedoes."
+
+outfit "Teciimach Bay"
+	plural "Teciimach Bays"
+	category "Secondary Weapons"
+	licenses
+		Remnant
+	cost 583000
+	thumbnail "outfit/teciimach bay"
+	"mass" 9
+	"outfit space" -13
+	"weapon capacity" -13
+	"gun ports" -1
+	"teciimach canister capacity" 9
+	weapon
+		sprite "projectile/emp torpedo"
+			"frame rate" 2
+			"no repeat"
+		sound "torpedo"
+		ammo "Teciimach Canister"
+		icon "icon/emp"
+		"fire effect" "emp torpedo fire" 5
+		"hit effect" "nuke explosion"
+		"hit effect" "emp ring" 30
+		"hit effect" "emp spark" 20
+		"die effect" "tiny explosion"
+		cluster
+		safe
```

Comment:
> "Well why doesn't anyone else?"

Primary answer would be "because no-one else uses EMPs." And even if they did, this would require careful synchronization that requires it to be connected somehow. Ex. If some Remnant ships went rogue and attacked other Remnant, they would be subject to each-other's EMPs, since they aren't sharing the deployment info required for this synchronization. The wave-cancelling we see in present day is just one possibility. There's a *lot* of other ways to mitigate electrical damage; and Star Trek is filled with techno-babble incidents of people synchronizing shields to effects in order to save the day; typically either an effect they control, or one they've had the chance to study at length.

> Perhaps the Remnant create an outfit alongside this weapon which provides a great deal of ion resistance? Basically a shield generator but for fighting ionization. All Remnant craft which are expected to be fielded alongside a Teciimach would then be equipped with this device, and even some ships that aren't expecting to be fighting alongside a Teciimach would likely use this device due to it helping with resisting ion storms.

I agree that the justification makes perfect sense, but I'd just like to point out that them having an outfit that can largely negate the effects of an EMP hit seems quite a bit more advanced / high-tech than simply having developed internal mechanisms for mitigating the effects of a weapon that they launched in the first place. This is why I went with the `safe` attribute instead of resistances. If you're good with resistances, though, I can do that.

I would opt for a distributed approach, though: 
- Some ionization protection on the ships themselves, fitting in with the lore of semi-organic ships that don't rely on classic electronics as much. They aren't any better than other ships at *removing* ionization, they just don't get as much of it to begin with. This would be scaled, so older (more electrical) ships would have less protection than newer ships.
- Buffed ionization resistance on the existing outfits that already have it. The resistance values are, so far as I can tell, too low for anyone to consider them as anything other than a slightly appreciated footnote.

Reply:
> I agree that the justification makes perfect sense, but I'd just like to point out that them having an outfit that can largely negate the effects of an EMP hit seems quite a bit more advanced / high-tech than simply having developed internal mechanisms for mitigating the effects of a weapon that they launched in the first place.

I don't see how partially negating an EMP hit any more advanced than COMPLETELY negating one. Even if the explanation is something to the effect of "They do a lot with a little," gameplay wise a complete negation of something is more advanced than a partial negation of it.

> Some ionization protection on the ships themselves...

I'll give you 0.1 at the *most*. But really, given the numbers below, I'd rather not even bother with ion protection.

> Buffed ionization resistance on the existing outfits that already have it.

Ships vs the ionization of the Teciimach (65 ion), time for ionization to decay to various values / total energy lost by that point when equipped with nothing or one of the ion resistance Remnant outfits (Crystal Capacitor, Millennium Cell, Epoch Cell, Aeon Cell).

Ion value | 0 resistance | 0.01 resistance | 0.025 resistance | 0.085 resistance | 0.14 resistance
---|---|---|---|---|---
60 | 8 frames / 502 energy | 8 frames / 502 energy | 8 frames / 502 energy | 8 frames / 500 energy | 7 frames / 439 energy
50 | 27 frames / 1545 energy | 26 frames / 1492 energy | 26 frames / 1487 energy | 23 frames / 1321 energy | 21 frames / 1209 energy
40 | 49 frames / 2528 energy | 48 frames / 2478 energy | 47 frames / 2424 energy | 42 frames / 2174 energy | 38 frames / 1976 energy
30 | 77 frames / 3502 energy | 76 frames / 3449 energy | 73 frames / 3327 energy | 65 frames / 2973 energy | 59 frames / 2708 energy
20 | 118 frames / 4515 energy | 114 frames / 4387 energy | 110 frames / 4240 energy | 95 frames / 3714 energy | 84 frames / 3328 energy
10 | 187 frames / 5508 energy | 179 frames / 5329 energy | 168 frames / 5083 energy | 138 frames / 4341 energy | 119 frames / 3845 energy
5 | 256 frames / 6004 energy | 239 frames / 5764 energy | 219 frames / 5455 energy | 169 frames / 4569 energy | 142 frames / 4016 energy
1 | 416 frames / 6400 energy | 348 frames / 6052 energy | 295 frames / 5664 energy | 204 frames / 4670 energy | 166 frames / 4086 energy

And here's a table of the final results with comparisons against no resistance. Only taking the decay time and energy lost to get to an ionization of 1 since the last bit of ion takes a while to decay off, and gameplay wise isn't hurting you much. 

I've also included a rough equivalent amount of ion protection one would need to receive the same reduction in energy loss for the sake of comparing protection to resistance. Note that the equivalent protection for any level of resistance depends on the amount of ionization that was applied; resistance is actually better against smaller values of ionization (such as those received during an ion storm), whereas protection is the same flat buff to energy loss across all ionization values. Resistance being better at low values can be observed by how at different resistance levels, the amount of energy lost near the start is very close, then as time goes on the resistances deviate from one another in total energy lost.

Resistance | Decay time (frames) | % better than 0 | Energy lost | % better than 0 | Protection equivalent
---|---|---|---|---|---
0 | 416 | | 6400 | | 0
0.01 | 348 | 16.3% | 6052 | 5.4% | 0.057
0.025 | 295 | 29.1% | 5664 | 11.5% | 0.130
0.085 | 204 | 51.0% | 4670 | 27.0% | 0.370
0.14 | 166 | 60.1% | 4086 | 36.2% |  0.567

I compare both decay times and energy lost because the new effect of ionization jamming weapons means that decay time becomes just as if not more important than total energy lost, as decaying ionization quicker means getting jammed less.

Basically, I don't think any existing outfits need buffed. If a Starling with an Epoch cell or an Albatross with an Aeon Cell can recover more than twice as fast as whatever Korath ship it's attacking, that's pretty good already.

> Having a specific anti-ionization outfit would be a separate unlock requiring more missions, so would be a separate PR for a later date.

I'm fine with that if we don't use `safe` here. Given the numbers above, I would say that an ion resistance of 0.03 on a relatively small dedicated outfit (say, 17 tons) might work. That's roughly double the resistance per ton of the existing Remnant outfits. Although unlike the passive ion resistance of those outfits, I'd say giving this dedicated outfit an energy cost would make sense, say `"ion resistance energy" 3`.

Reply:
"ion resistance energy" of 3 on an "ion resistance" of 0.03 essentially means it'll remove all the energy the 0.03 ionisation would have in its lifetime immediately. That's not super useful. It removes the same energy over a much shorter time (which is arguably worse). It does have the benefit of reducing the amount of weapon jams the ship is likely to experience but that's it. Doubling the resistance per ton of the outfits that are also power generators or storage doesn't seem like enough to me. How would installing another crystal capacitor compare to this outfit in terms of actual performance against ion storms and various ionising weapons?

Reply:
The other problem would be that no amount of ionization resistance or protection will do anything for the rest of the Teciimach's effects. The Shield damage and energy hit are just as big a problem as the ionization.

Reply:
> It does have the benefit of reducing the amount of weapon jams the ship is likely to experience but that's it.

That's the intention. Preventing jamming can be a big boost, so I wouldn't say that it's "not super useful."

> How would installing another crystal capacitor compare to this outfit in terms of actual performance against ion storms and various ionising weapons?

Well crystal capacitors already consume 0.4 energy per frame, so you could perhaps imagine that as crystal capacitors having roughly half the ion resistance per ton in exchange for being roughly twice as energy efficient with bleeding ionization. Although crystal capacitors constantly drain energy, not just when bleeding ionization, so perhaps whether the reduced energy drain at a constant rate is better or worse is up to the user.

Reply:
> I don't see how partially negating an EMP hit is any more advanced than COMPLETELY negating one. Even if the explanation is something to the effect of "They do a lot with a little," gameplay wise a complete negation of something is more advanced than a partial negation of it.

Because it is selectively negating *only* the effects of a a detonation where the launching ship knows, to the nanosecond, the exact time of launch, time of detonation, exact characteristics of said detonation in every respect, etc. Any other circumstance, such as the same weapon launched by a hostile ship? Zero protection. A sort of similar weapon from another faction? Also zero protection. 

So swapping the "safe" function here for some very minor ionization resistance is trading the ability to use this weapon effectively for protection against *every* weapon that makes use of ionization, no matter the source, mechanism, or type.

----

Since it seems to be the capability to synchronize shields that is the problem, would changing the description make it more palatable? For instance, stating that the weapon burst has a mechanism to control the directionality of its emissions to only affect enemy ships? I'd be fine with that.

Reply:
> And, again, I'd like to point out that everyone is focusing really heavily on the ionization part, and ignoring that this weapon actually does do shield damage, energy damage, and hull damage. No amount of ionization resistances are going to mitigate those other aspects.

I don't think that actually helps your argument that allied ships being able to prepare for the blast enables the `safe` functionality.

> to the nanosecond, the exact time of launch, time of detonation, exact characteristics of said detonation in every respect

That's a lot of very precise details. I have trouble believing that even the firing ship would know when the projectile was launched *to the nanosecond*, let alone knowing when it will detonate to such precision. And implies a high level of predictability to these missiles. If "shield polarisation" can negate all of these things so completely because so much is known about the detonation, why should it not be possible to adapt to incoming enemy fire with sufficient analysis?

If the shielding is the justification for the `safe`ness why does it persist even when shields are fully drained? What if I have a ship with no shields?

> Since it seems to be the capability to synchronize shields that is the problem, would changing the description make it more palatable? For instance, stating that the weapon burst has a mechanism to control the directionality of its emissions to only affect enemy ships?

Yes, please.

Reply:
Note: The mechanism has been changed, so it is now an actively shaped emission pattern rather than synchronization with the launching ship. See recent commits for specifics.

Reply:
Quick note, EMP can scale upwards, so unless you've protected the ship in question to withstand anything higher than a Lightning Strike, it's unlikely your vessel could withstand a dedicated EMP weapon.

That, and inactive electronics aren't as badly affected as active ones.

Reply:
I think this aspect as been more or less resolved, so marking it as such for clarity.